File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.swift
import Foundation

func solution(_ s:String) -> Bool
{
    var stack : [Character] = []
    
    for i in s {
        if i == "(" {
            stack.append("(")
        } else if i == ")" {
            if stack.isEmpty {
                return false
            }
            stack.popLast()
        }
    }
    if stack.count == 0 {
        return true
    } else {
        return false
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    
    left = []
    
    for b in s:
        if b == "(":
            left.append(b)
        else:
            if left:
                left.pop()
            else:
                return False
            
    if not left:
        return True
    else:
        return False